bookkeeper-server/src/main/java/org/apache/bookkeeper/client/SyncCallbackUtils.java

```diff
@@ -0,0 +1,287 @@
+            finish(rc, lh, future);
+        }
+    }
+
+    static class SyncDeleteCallback implements AsyncCallback.DeleteCallback {
+
+        private final CompletableFuture<Void> future;
+
+        public SyncDeleteCallback(CompletableFuture<Void> future) {
+            this.future = future;
+        }
+
+
+        /**
+         * Delete callback implementation for synchronous delete call.
+         *
+         * @param rc
+         *            return code
+         * @param ctx
+         *            optional control object
+         */
+        @Override
+        @SuppressWarnings("unchecked")
+        public void deleteComplete(int rc, Object ctx) {
+            finish(rc, null, future);
+        }
+    }
+
+    static class LastAddConfirmedCallback implements AsyncCallback.AddLacCallback {
+        static final LastAddConfirmedCallback INSTANCE = new LastAddConfirmedCallback();
+        /**
+         * Implementation of callback interface for synchronous read method.
+         *
+         * @param rc
+         *          return code
+         * @param lh
+         *          ledger identifier
+         * @param ctx
+         *          control object
+         */
+        @Override
+        public void addLacComplete(int rc, LedgerHandle lh, Object ctx) {
+            if (rc != BKException.Code.OK) {
+                LOG.warn("LastAddConfirmedUpdate failed: {} ", BKException.getMessage(rc));
+            } else {
+                if (LOG.isDebugEnabled()) {
+                    LOG.debug("Callback LAC Updated for: {} ", lh.getId());
+                }
+            }
+        }
+    }
+
+    static class SyncReadCallback implements AsyncCallback.ReadCallback {
+        /**
+         * Implementation of callback interface for synchronous read method.
+         *
+         * @param rc
+         *          return code
+         * @param lh
+         *          ledger handle
+         * @param seq
+         *          sequence of entries
+         * @param ctx
+         *          control object
+         */
+        @Override
+        @SuppressWarnings("unchecked")
+        public void readComplete(int rc, LedgerHandle lh,
+                                 Enumeration<LedgerEntry> seq, Object ctx) {
+            finish(rc, seq, (CompletableFuture<Enumeration<LedgerEntry>>)ctx);
+        }
+    }
+
+    static class FutureReadResult extends CompletableFuture<Iterable<org.apache.bookkeeper.client.api.LedgerEntry>>
+        implements AsyncCallback.ReadCallback {
+
+        /**
+         * Implementation of callback interface for read method of {@link ReadHandle}.
+         *
+         * @param rc
+         *          return code
+         * @param lh
+         *          ledger handle
+         * @param seq
+         *          sequence of entries
+         * @param ctx
+         *          control object
+         */
+        @Override
+        @SuppressWarnings("unchecked")
```

Comment:
why @SuprressWarnings?

Reply:
because it is an unchecked cast from Iterable<legacy LedgerEntry> and the new Iterable<new LedgerEntry>